src/NServiceBus.NHibernate/SharedSession/NHibernateStorageSession.cs

```diff
@@ -79,5 +77,26 @@ protected override void Setup(FeatureConfigurationContext context)
             Installer.RunInstaller = context.Settings.Get<bool>("NHibernate.Common.AutoUpdateSchema");
             Installer.configuration = configuration;
         }
+
+        static bool DisableConnectionSharing(FeatureConfigurationContext context, Configuration configuration)
+        {
+            return context.Settings.GetOrDefault<bool>("NServiceBus.Features.SqlServerTransportFeature")
+                   && context.Settings.GetOrDefault<bool>(typeof(Outbox).FullName)
+                   && (SqlServerDriver(configuration) || SqlServerDialect(configuration));
```

Comment:
I don't like it either. This makes the SQL/NH/Outbox mode not require `NHibernateStorageContext` in order for user to store data.